src/bcf/record.rs

```diff
@@ -157,6 +157,96 @@ impl Record {
         }
     }
 
+    /// Set the given filters to the FILTER column.
+    ///
+    /// Note that the given strings must have been registered in the header and it will be
+    /// automatically converted to the corresponding integer value for being added.
+    ///
+    /// Setting an empty slice removes all filters.
+    ///
+    /// # Args
+    /// - `val` - The corresponding filter string value.
+    pub fn set_filters(&mut self, vals: &[Vec<u8>]) -> Result<(), FilterWriteError> {
```

Comment:
I wonder if we should allow to set filters by id, and get the ID first. I.e., expose a method `bcf::HeaderView::id(&[u8]) -> bcf::header::ID`. And then have here `set_filters(&mut self, vals: &[bcf::header::ID])`. What do you think? This also guarantees that only available IDs are provided. In a way, this supports the Rust paradigm of trying to fail at compile time already.

Reply:
Hm, think we should offer both an integer-based and a convenience interface. However, I don't see how this connects to failing at compile time. The integer-based API will allow for handling individual string-to-id resolution errors (which is useful) but nothing more.

Reply:
After thinking about this some more, I agree that we should rely on bcftools integers here. Adjusting accordingly.

Reply:
I meant that since the method `HeaderView::id` is the only one to get `bcf::header::ID`, the API kind of enforces that user cannot easily put an invalid ID into set_filters.